features/marshal/marshalto.go

```diff
@@ -50,8 +53,18 @@ func (p *marshal) GenerateFile(file *protogen.File) bool {
 	return p.once
 }
 
+// need different names to generate helpers for marshal and marshal_strict features independently
+func (p *marshal) encodeVarintMeth() string {
+	switch {
+	case p.strict:
+		return "encodeVarintForStrict"
+	default:
+		return "encodeVarint"
+	}
+}
+
 func (p *marshal) GenerateHelpers() {
-	p.P(`func encodeVarint(dAtA []byte, offset int, v uint64) int {`)
+	p.P(`func `, p.encodeVarintMeth(), `(dAtA []byte, offset int, v uint64) int {`)
```

Comment:
I don't understand why you need a specific name for `encodeVarint`. The implementation is identical to the previous, so why name it differently?

Reply:
As i commented on 56 line in order to `generate helpers for marshal and marshal_strict features independently`.
E.g. if we want to use both `marshal` and `marshal_strict` features then `encodeVarint` impl would be generated twice that will cause to compilation error.

Reply:
Sorry for the delay getting back to this, you caught me in holiday. We need to do better than that for this feature: let's check during codegen whether both `marshal` and `marshal_strict` are being requested and handle that case by generating the `encodeVarint` helpers only once.

Reply:
Sorry for long delay! Returned to this PR. Watch please, added duplicate checker for helper functions to avoid duplicates among helper functions names which are generated by different features in the same file.